scripts/irods/test/test_iadmin.py

```diff
@@ -2226,6 +2226,28 @@ def test_moduser_password(self):
             with session.make_session_for_existing_user(self.username, pw1, host, self.admin.zone_name):
                 pass
 
+        # Change password to 42 bytes (should succeed)
+        pw3 = '123456789012345678901234567890123456789012'
+        self.admin.assert_icommand(['iadmin', 'moduser', self.username, 'password', pw3])
+        with session.make_session_for_existing_user(self.username, pw3, host, self.admin.zone_name):
+            pass
+
+        # Make sure old password does not work
+        with self.assertRaises(Exception):
+            with session.make_session_for_existing_user(self.username, pw2, host, self.admin.zone_name):
+                pass
+
+        # Change password to 43 bytes (should fail)
+        pw4 = '1234567890123456789012345678901234567890123'
+        self.admin.assert_icommand(['iadmin', 'moduser', self.username, 'password', pw4], 'STDERR', 'PASSWORD_EXCEEDS_MAX_SIZE')
+
+        # Change password to 45 bytes (should fail)
+        pw4 = '123456789012345678901234567890123456789012345'
+        self.admin.assert_icommand(['iadmin', 'moduser', self.username, 'password', pw4], 'STDERR', 'PASSWORD_EXCEEDS_MAX_SIZE')
+
+        # Change password to 46 bytes (should fail)
+        pw4 = '1234567890123456789012345678901234567890123456'
+        self.admin.assert_icommand(['iadmin', 'moduser', self.username, 'password', pw4], 'STDERR', 'PASSWORD_EXCEEDS_MAX_SIZE')
```

Comment:
Consider adding an assertion following this that proves the failed calls to modify the user's password don't result in breaking the user's ability to execute commands. The assertion can do an `ils` or something else entirely.

Reply:
I added some code to verify that a session can be created with the last "good" password and not with the "bad" passwords.

Reply:
I added an "ils" command after creating the session for the last good password.